### PR TITLE
Upgrade to web3 1.7.0, upgrade wyvern-js and wyvern-schemas

### PR DESCRIPTION
**Pull request by sunwrobert**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
+    "target": "es6",
     "lib": ["dom", "esnext"],
     "allowSyntheticDefaultImports": true,
     "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "noImplicitReturns": true,
     "noErrorTruncation": true,
     "noImplicitThis": false,
-    "noUnusedLocals": true,
+    "noUnusedLocals": false,
```
Comment (Meemaw) [inline tsconfig.json:17]: Can we keep this on `true`. Unused locals make no sense, and we should just out remove them where applicable.

File: package.json [hunk 2]
```diff
@@ -17,11 +17,12 @@
...
   ],
   "scripts": {
-    "build": "tsc",
+    "abi-type-gen": "yarn run typechain --target=web3-v1 src/abi/*.json --out-dir=src/typechain/contracts",
+    "build": "yarn abi-type-gen && tsc",
     "check-types": "tsc --noEmit --project tsconfig.json",
     "coverage-report": "nyc report --reporter=text-lcov | coveralls",
     "docs-build": "typedoc --out docs src/index.ts",
-    "eslint:check": "eslint . --max-warnings 0 --ext .js,.ts",
+    "eslint:check": "eslint src --max-warnings 0 --ext .js,.ts",
```
[the remark this answers is not in the document]
Comment (sunwrobert) [reply package.json:25]: can change, just need to ignore `lib`

File: tsconfig.json [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
   "include": [
     "./src/**/*",
-    "./node_modules/web3-typescript-typings/index.d.ts",
     "./node_modules/types-bn/index.d.ts",
     "./node_modules/types-ethereumjs-util/index.d.ts"
   ],
   "compilerOptions": {
     "outDir": "lib",
-    "target": "es5",
+    "target": "es6",
```
[the remark this answers is not in the document]
Comment (sunwrobert) [reply tsconfig.json:9]: Can try removing

File: tsconfig.json (hunk 1 above)
Comment (sunwrobert) [reply tsconfig.json:17]: Adding a comment that it's needed due to auto generated files

**Review by Meemaw — APPROVED** [review]

**Comment by bujoralexandru** [comment]
image
image

I think my issue above is related to this pull request, so can this be integrated faster, please?

**Comment by sunwrobert** [comment]
> image image
> 
> I think my issue above is related to this pull request, so can this be integrated faster, please?

Later this month is the plan!

**Pull request merged**